packages/material/test/renderers/MaterialOneOfRenderer.test.tsx

```diff
@@ -0,0 +1,404 @@
+            },
+            {
+              title: 'Number',
+              type: 'number'
+            }
+          ]
+        }
+      }
+    };
+    const uischema: ControlElement = {
+      type: 'Control',
+      label: 'Value',
+      scope: '#/properties/value'
+    };
+    store.dispatch(Actions.init({data: undefined}, schema, uischema));
+    wrapper = mount(
+      <Provider store={store}>
+        <MaterialOneOfRenderer schema={schema} uischema={uischema} />
+      </Provider>
+    );
+    const input = wrapper.find('input').first();
+    input.simulate('change', { target: { value: 'test' }});
+    wrapper.update();
+    expect(store.getState().jsonforms.core.data).toEqual({
+      value: 'test'
+    });
+  });
+
+  it('should add an item within an array', () => {
+    const schema = {
+      type: 'object',
+      properties: {
+        oneOrMoreThings: {
+          oneOf: [
+            {
+              $ref: '#/definitions/thing'
+            },
+            {
+              $ref: '#/definitions/thingArray'
+            }
+          ]
+        }
+      },
+      definitions: {
+        thing: {
+          title: 'Thing',
+          type: 'string'
+        },
+        thingArray: {
+          title: 'Things',
+          type: 'array',
+          items: {
+            $ref: '#/definitions/thing'
+          }
+        }
+      }
+    };
+    const resolvedSchema = {
+      type: 'object',
+      properties: {
+        thingOrThings: {
+          oneOf: [
+            {
+              title: 'Thing',
+              type: 'string',
+            },
+            {
+              title: 'Things',
+              type: 'array',
+              items: {
+                title: 'Thing',
+                type: 'string'
+              }
+            }
+          ]
+        }
+      }
+    };
+    const uischema: ControlElement = {
+      type: 'Control',
+      scope: '#/properties/thingOrThings'
+    };
+
+    const store = initStore();
+    store.dispatch(Actions.init({ data: {}, schema, uischema}));
+    const promise = Promise.resolve({ resolved: resolvedSchema});
+    const mock = jest.spyOn(JsonRefs, 'resolveRefs');
+    mock.mockReturnValue(promise);
+
+    wrapper = mount(
+      <Provider store={store}>
+        <JsonForms schema={schema} uischema={uischema} />
+      </Provider>
+    );
+
+    return promise.then(() => {
+      // expect(wrapper.state())
+      wrapper.update();
+
+      // select oneOf-1
+      let buttons = wrapper.find('button');
+      buttons.at(1).simulate('click');
+      wrapper.update();
+
+      // confirm dialog
+      buttons = wrapper.find('button');
+      buttons.at(3).simulate('click');
+      wrapper.update();
+
+      // click add button
+      buttons = wrapper.find('button');
+      const addButton = buttons.at(2);
+      const nrOfRowsBeforeAdd = wrapper.find('tr');
+      addButton.simulate('click');
+      addButton.simulate('click');
+
+      // check if row has been added
```

Comment:
this should say 'check if row**s** have been added as you added 2

Reply:
Depends on the view point: initially we already have two rows (header + no data), if we add two items, one is replacing the no data one, hence there's one row more than initially. But I can update if you want, if 'rows' seems to make more sense to you.

Reply:
I'm fine with both, my thinking was that you had no data, and add two entries, but I see what you meant.